controller/controller.go

```diff
@@ -387,21 +387,20 @@ func (c *ObjectStorageController) runController(ctx context.Context) {
 							}
 
 							uuid := d.Object.(metav1.Object).GetUID()
-							c.OpLock(uuid)
-							defer c.OpUnlock(uuid)
-							
+
 							// If an update to the k8s object happens before add has succeeded
-							if op, ok := c.opMap[uuid]; ok {
+							if op, ok := c.opMap.Load(uuid); ok {
 								if _, ok := op.(updateOp); ok {
-									return fmt.Errorf("cannot add already added object: %s", key)
+									err := fmt.Errorf("cannot add already added object: %s", key)
+									return err
 								}
 							}
-							c.opMap[uuid] = addOp{
+							c.opMap.Store(uuid, addOp{
 								Object:  d.Object,
 								AddFunc: &add,
 								Key:     key,
 								Indexer: indexer,
-							}
+							})
```

Comment:
Could we combine this by using `c.opMap.LoadOrStore` /nit

Reply:
Sure